app/presenters/document_presenter.rb

```diff
@@ -129,22 +125,34 @@ def default_date_metadata
     caption = first_edition? ? "Published" : "Updated"
 
     {
-      caption => published_at,
+      caption.downcase => {
+        label: caption,
+        values: public_updated_at,
+      }
     }
   end
 
+  def extra_date_metadata
+    keys = finder.date_facets.map(&:key)
+    get_metadata(keys)
+  end
+
   def first_edition?
     change_history.size <= 1
   end
 
   def filterable_metadata
-    finder.text_facets.select(&:filterable)
-          .each_with_object({}) { |facet, hash| hash[facet.key] = self.send(facet.key) }
+    keys = finder.text_facets.select(&:filterable).map(&:key)
+    get_metadata(keys)
   end
 
   def extra_metadata
-    finder.text_facets.reject(&:filterable)
-          .each_with_object({}) { |facet, hash| hash[facet.name] = self.send(facet.key) }
+    keys = finder.text_facets.reject(&:filterable).map(&:key)
+    get_metadata(keys)
+  end
+
+  def get_metadata(keys)
+    document.details.metadata.to_h.stringify_keys.slice(*keys)
```

Comment:
since this is called several times by other various methods, might be a good idea to store the unsliced (but stringified) metadata hash as an instance var.